[Aaron Li] ip

aaronliwh opened the pull request:
## DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
 - text-based
 - easy to learn
 - ~~FAST~~ SUPER FAST to use
 
 All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)  
 
#
If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

aaronliwh committed:
Level 1: Implemented Greet,Echo and Exit

aaronliwh committed:
Level 2: Implementd Add,List

aaronliwh committed:
Level 3: Implemented Marking tasks

aaronliwh committed:
Level 4: Implemented ToDos, Events, Deadlines

aaronliwh committed:
Implemented Automated Text UI Testing

aaronliwh committed:
Implemented basic error handling

aaronliwh committed:
Implemented Delete

aaronliwh committed:
Added more exception handling

aaronliwh committed:
Added javadocs

aaronliwh committed:
Added more testcases

aaronliwh committed:
Minor bug fixes

aaronliwh committed:
Added Save

aaronliwh committed:
Added Dates and Times

aaronliwh committed:
Merge branch 'branch-Level-7'

aaronliwh committed:
Merging Level-8

aaronliwh committed:
Merge remote-tracking branch 'ip/add-gradle-support'

aaronliwh committed:
Add A-MoreOOP

aaronliwh committed:
Add A-Packages

aaronliwh committed:
Add A-JUnit

aaronliwh committed:
Added Javadocs

aaronliwh committed:
Edit code to coding standard

aaronliwh committed:
Updated coding standard

aaronliwh committed:
Add Find

aaronliwh committed:
Merge branch 'branch-A-JavaDoc'

aaronliwh committed:
merge branch-A-CodingStandard

aaronliwh committed:
merge branch-Level9

aaronliwh committed:
Add CheckStyle

aaronliwh committed:
Add basic GUI

aaronliwh committed:
Add Duke GUI

aaronliwh committed:
Merge branch 'branch-Level-10'

RussellDash332 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,24 @@
+
+    public String getResponse(String input) {
+        try {
+            Command c = Parser.parse(input);
```
Since you have a `parseTask` method somewhere in this repo, do you think it's a good idea to rename this method to `parseInput`?

RussellDash332 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+                LocalDate dateEvent = getDate(stringDateEvent);
+                return new AddCommand("event",nameEvent,dateEvent);
+            case "mark" :
+                int indexMark = getNumber(input);
```
(Analogously for the cases below)
Perhaps `markIndex` would be a more sensible naming since it really sounds like a noun?

darrentde replied:
I agree with @RussellDash332 suggestion that `markIndex` may be a better name

darrentde replied:
I agree with @RussellDash332 suggestion that `markIndex` may be a better name

RussellDash332 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,98 @@
+     * Returns the number of tasks in the list
+     * @return the size of the List of Tasks
+     */
+    public int size(){
```
What do you think of `getSize` instead of `size`? It might be clearer to even more unfamiliar readers

RussellDash332 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,79 @@
+     * Prints the ending message
+     * @return
+     */
+    public String goodbyeMessage() {
+        return "Bye. Hope to see you again soon!";
+    }
+
+    /**
+     * Reads the User's input
+     * @return The String representation of the user's input
+     */
+
+
+    /**
+     * Prints the appropriate message for adding a Task
+     * @param task The task added
+     * @param size The current number of tasks
+     */
+    public String showAddCommand(Task task,int size) {
```
Both of these methods are returning a String, but only the latter is a verb. Should you rename `goodbyeMessage` to a verb as well?

RussellDash332 reviewed (commented):
LGTM! Very little nits to deal with since my suggestions so far are somewhat not an urgent one but probably good to consider 👍🏻

Loving the GUI resources as well 💪🏻

darrentde commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,98 @@
+package duke;
```
Perhaps you could start to create more specified packages for your files since you have already done it for command?

darrentde reviewed (commented):
LGTM. In general your code meets the naming conventions. 🤩

RachelChua commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+        switch (input[0]) {
+            case "bye":
+                return new EndCommand();
+            case "list":
```
Small issue but based on Java coding standard guidelines, I think there should not be indentations for case clauses. 😃

RachelChua commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,117 @@
+        char type = (data.charAt(0));
+        boolean done = (data.charAt(1)) == ('Y');
+        Task task = null;
+        if(type == 'T'){
```
Seems to have a missing space between if and the round opening bracket, as well as the closing round bracket and the curly brackets.

RachelChua commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,98 @@
+     */
+    public String printTaskList(){
+        String returnString = "";
+        for(int i = 0; i<taskList.size(); i++){
```
Should there be spaces between the brackets?

RachelChua commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,24 @@
+/**
+ * Encapsulates a command.
+ */
+public abstract class Command {
```
I like your abstraction for the command class!

RachelChua reviewed (approved):
Just a few minor coding standards issues but overall, I think that your OOP and abstraction is done well! Good job! 👍

aaronliwh committed:
Add one test assert

aaronliwh committed:
Change one method name

aaronliwh committed:
Add Assertions

Added assertions to detect possible
unexpected behaviours. Asserted
non-negative indexes for TaskList
and non-null dates for Events and
Deadlines

aaronliwh committed:
Improve code quality

Serveral method and variable names do not conform to the Java
coding standards. Additionally, there are some minor formting errors
such as lack of spacing and incorrect indentation levels. Fixing
these will make the code more standardised and readable.

aaronliwh committed:
Merge pull request #3 from ALWHgithub/branch-A-Assertions

Branch a assertions

aaronliwh committed:
Merge branch 'branch-A-CodeQuality'

aaronliwh committed:
Add ability to detect duplicates

Duplicate tasks are redundant and cause the tasklist to be
needlessly complicated. Detecting duplicates acording to the
toString value of the tasks

aaronliwh committed:
Fix bugs with Duke

Change the following
* Program exits when "Bye" is entered
* List displays all task in the same row without indentation

aaronliwh committed:
Add more asserts

aaronliwh committed:
Minor UI changes

aaronliwh committed:
Polish GUI

aaronliwh committed:
Add CI

aaronliwh committed:
Merge pull request #4 from ALWHgithub/branch-C-DetectDuplicates

Add ability to detect duplicates

aaronliwh committed:
Merge detecting duplicates

aaronliwh committed:
Add ReadMe

aaronliwh committed:
Add more assert

aaronliwh committed:
Improve GUI

aaronliwh committed:
Add checkstyle

aaronliwh committed:
Updated Ui.png

aaronliwh committed:
Change gradle.yml
